applications/nrf5340_audio/src/audio/streamctrl_unicast_server.c

```diff
@@ -0,0 +1,722 @@
+	}
+}
+
+/* Handle Bluetooth LE audio events */
+static void le_audio_msg_sub_thread(void)
+{
+	int ret;
+	uint32_t pres_delay_us;
+	uint32_t bitrate_bps;
+	uint32_t sampling_rate_hz;
+	const struct zbus_channel *chan;
+
+	while (1) {
+		ret = zbus_sub_wait(&le_audio_evt_sub, &chan, K_FOREVER);
+		ERR_CHK(ret);
+
+		struct le_audio_msg msg;
+
+		ret = zbus_chan_read(chan, &msg, ZBUS_READ_TIMEOUT_MS);
+		ERR_CHK(ret);
+
+		uint8_t event = msg.event;
+
+		LOG_DBG("Received event = %d, current state = %d", event, strm_state);
+
+		switch (event) {
+		case LE_AUDIO_EVT_STREAMING:
+			LOG_DBG("LE audio evt streaming");
+
+			if (strm_state == STATE_STREAMING) {
+				LOG_DBG("Got streaming event in streaming state");
+				break;
+			}
+
+			audio_system_start();
+			stream_state_set(STATE_STREAMING);
+			ret = led_blink(LED_APP_1_BLUE);
+			ERR_CHK(ret);
+
+			break;
+
+		case LE_AUDIO_EVT_NOT_STREAMING:
+			LOG_DBG("LE audio evt not_streaming");
+
+			if (strm_state == STATE_PAUSED) {
+				LOG_DBG("Got not_streaming event in paused state");
+				break;
+			}
+
+			stream_state_set(STATE_PAUSED);
+			audio_system_stop();
+			ret = led_on(LED_APP_1_BLUE);
+			ERR_CHK(ret);
+
+			break;
+
+		case LE_AUDIO_EVT_CONFIG_RECEIVED:
+			LOG_DBG("Config received");
+
+			ret = unicast_server_config_get(&bitrate_bps, &sampling_rate_hz, NULL);
+			if (ret) {
+				LOG_WRN("Failed to get config: %d", ret);
+				break;
+			}
+
+			LOG_DBG("Sampling rate: %d Hz", sampling_rate_hz);
+			LOG_DBG("Bitrate: %d bps", bitrate_bps);
+
+			break;
+
+		case LE_AUDIO_EVT_PRES_DELAY_SET:
+			LOG_DBG("Set presentation delay");
+
+			ret = unicast_server_config_get(NULL, NULL, &pres_delay_us);
+			if (ret) {
+				LOG_ERR("Failed to get config: %d", ret);
+				break;
+			}
+
+			ret = audio_datapath_pres_delay_us_set(pres_delay_us);
+			if (ret) {
+				LOG_ERR("Failed to set presentation delay to %d", pres_delay_us);
+				break;
+			}
+
+			LOG_INF("Presentation delay %d us is set by initiator", pres_delay_us);
+
+			break;
+
+		case LE_AUDIO_EVT_NO_VALID_CFG:
+			LOG_WRN("No valid configurations found, will disconnect");
+
+			ret = bt_mgmt_conn_disconnect(msg.conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+			if (ret) {
+				LOG_ERR("Failed to disconnect: %d", ret);
+			}
+
+			break;
+
+		default:
+			LOG_WRN("Unexpected/unhandled le_audio event: %d", event);
+
+			break;
+		}
+
+		STACK_USAGE_PRINT("le_audio_msg_thread", &le_audio_msg_sub_thread_data);
+	}
+}
+
+/* Handle content control events */
+static void content_control_msg_sub_thread(void)
+{
+	int ret;
+
+	const struct zbus_channel *chan;
+
+	while (1) {
+		ret = zbus_sub_wait(&content_control_evt_sub, &chan, K_FOREVER);
+		ERR_CHK(ret);
+
+		struct content_control_msg msg;
+
+		ret = zbus_chan_read(chan, &msg, ZBUS_READ_TIMEOUT_MS);
+		ERR_CHK(ret);
+
+		uint8_t event = msg.event;
+
+		LOG_DBG("Received event = %d, current state = %d", event, strm_state);
+
+		switch (event) {
+		case MEDIA_PLAY:
+			break;
```

Comment:
I have a question in here:
since in would we need to send event in line 287, and subscribe that in here, and handling the play/pause event?

Reply:
I have removed this entire thread.